### PR TITLE
Add Thwaites 4km test group to landice core

### PR DESCRIPTION
**Pull request by matthewhoffman**
This merge adds the Thwaites variable resolution 4-14km test group to the landice core.  This test group uses a pre-existing mesh and initial condition file and provides decomposition and restart tests.  The test structure follows closely to that used for the Greenland FO tests.  The tesst are added to the fo and full integration suites, and the test group is briefly described in the docs.  

In the future additional Thwaites tests cases could be added that actually build the mesh and initial condition.

### CODE REVIEW

**Comment by matthewhoffman** [comment]
The new tests pass on Badger and are both short enough to include in the standard regression suites:
```
00:38 PASS landice_thwaites_decomposition_test
00:57 PASS landice_thwaites_restart_test
```

File: compass/landice/tests/thwaites/restart_test/__init__.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        name = 'full_run'
+        step = RunModel(test_case=self, name=name, subdir=name, cores=cores,
+                        min_cores=min_cores, threads=1)
+        # modify the namelist options and streams file
+        step.add_namelist_file(
+            'compass.landice.tests.greenland.restart_test',
+            'namelist.full', out_name='namelist.landice')
+        step.add_streams_file(
+            'compass.landice.tests.greenland.restart_test',
+            'streams.full', out_name='streams.landice')
```
Comment (xylar) [inline compass/landice/tests/thwaites/restart_test/__init__.py:39]:
```suggestion
        step.add_namelist_file(
            'compass.landice.tests.thwaites.restart_test',
            'namelist.full', out_name='namelist.landice')
        step.add_streams_file(
            'compass.landice.tests.thwaites.restart_test',
            'streams.full', out_name='streams.landice')
```

Comment (xylar) [reply compass/landice/tests/thwaites/restart_test/__init__.py:39]: If this is on purpose, it would be better to have these files in some shared (framework) space so it's clear that they're shared between test cases.  But I suspect it's just a typo.

File: compass/landice/tests/thwaites/decomposition_test/__init__.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        Parameters
+        ----------
+        test_group : compass.landice.tests.thwaites
+            The test group that this test case belongs to
+
+        """
+        name = 'decomposition_test'
+        subdir = '{}'.format(name)
+        super().__init__(test_group=test_group, name=name, subdir=subdir)
```
Comment (xylar) [inline compass/landice/tests/thwaites/decomposition_test/__init__.py:25]: This is fine and probably made sense copying the `greenland` test cases.  Since `subdir` defaults to being the same as `name`, you can just do:
```suggestion
        super().__init__(test_group=test_group, name='decomposition_test')
```
If you like the clarity of giving the `subdir` explicitly, I would at least take out the unnecessary string formatting:
```suggestion
        name = 'decomposition_test'
        subdir = name
        super().__init__(test_group=test_group, name=name, subdir=subdir)
```

File: compass/landice/tests/thwaites/restart_test/__init__.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        Parameters
+        ----------
+        test_group : compass.landice.tests.thwaites
+            The test group that this test case belongs to
+
+        """
+        name = 'restart_test'
+        subdir = '{}'.format(name)
+        super().__init__(test_group=test_group, name=name, subdir=subdir)
```
Comment (xylar) [inline compass/landice/tests/thwaites/restart_test/__init__.py:25]: Same comments as for the `DecompositionTest`.

File: compass/landice/tests/thwaites/run_model.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        threads : int, optional
+            the number of threads the step will use
+
+        suffixes : list of str, optional
+            a list of suffixes for namelist and streams files produced
+            for this step.  Most steps most runs will just have a
+            ``namelist.landice`` and a ``streams.landice`` (the default) but
+            the ``restart_run`` step of the ``restart_test`` runs the model
+            twice, the second time with ``namelist.landice.rst`` and
+            ``streams.landice.rst``
```
Comment (xylar) [inline compass/landice/tests/thwaites/run_model.py:53]:
```suggestion
            for this step.  Most run steps will just have a
            ``namelist.landice`` and a ``streams.landice`` (the default) but
            the ``restart_run`` step of the ``restart_test`` runs the model
            twice, the second time with ``namelist.landice.rst`` and
            ``streams.landice.rst``
```
I'm guessing this typo is in all `RunModel` steps in `landice`. Sorry about that!

File: docs/users_guide/landice/test_groups/index.rst [hunk 5]
```diff
@@ -5,8 +5,8 @@ Test Groups
 
 For the time being, ``landice`` test groups that have been ported from legacy
 COMPASS to the ``compass`` package are those with test cases in the
-:ref:`landice_suite_sia_integration`.  We expect to port many more test cases
-in the coming months.
+:ref:`landice_suite_sia_integration` and many of the test cases in the
+:ref:`landice_suite_fo_integration`.
```
Comment (xylar) [inline docs/users_guide/landice/test_groups/index.rst:9]: This label needs to be added to `docs/users_guide/landice/suites.rst`:

```
...
(``restart_test``). Others ensure that the code produces the same result when
it is run on different numbers of cores (``decomposition_test``).

.. _landice_suite_fo_integration:

fo_integration test suite
-------------------------
...
```

**Comment by xylar** [comment]
I noticed a few things in my review.  The only one directly related to the PR is that there's not yet a section for the `thwaites` test group in the developer's guide.

Not directly related to this PR:
* the mesh link in the `greenland` test group (both in the user's and developer's guides) seems to be the old one (date stamp `150922`).
* You didn't make any changes to the developer's guide for `greenland` in #146, and that probably should also have been updated to reflect the new FO test cases.

Sorry I missed those 2.

**Comment by matthewhoffman** [comment]
@xylar , thanks for looking this over and providing me feedback.  I've fixed all the items and updated the missing parts of the docs.  Feel free to look it over again if you'd like, but you don't need to.

File: docs/users_guide/landice/test_groups/index.rst [hunk 6]
```diff
@@ -15,4 +15,5 @@ in the coming months.
    enthalpy_benchmark
    eismint2
    greenland
+   thwaites
```
Comment (xylar) [inline docs/users_guide/landice/test_groups/index.rst:18]: The corresponding addition is needed in the `docs/developers_guide/landice/test_groups/index.rst`.

Also, so far the tests have been alphabetical.  There's no reason that needs to be the case.  You may want to give them a more logical order if there is one.  That could be something for another PR, perhaps as more test groups get added

File: docs/developers_guide/landice/test_groups/thwaites.rst [hunk 7]
```diff
@@ -0,0 +1,64 @@
+.. _dev_landice_thwaites:
+
+thwaites
+=========
+
+The ``thwaites`` test group (:py:class:`compass.landice.tests.thwaites.thwaites`)
```
Comment (xylar) [inline docs/developers_guide/landice/test_groups/thwaites.rst:6]:
```suggestion
The ``thwaites`` test group (:py:class:`compass.landice.tests.thwaites.Thwaites`)
```

**Comment by xylar** [comment]
Two more small things.
